msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -2224,6 +2236,220 @@ public String processActionToggleDisplayExtendedDescription()
+      availabilityRestrictedNow = ((DiscussionTopic) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else if (forumItem instanceof DiscussionForum) {
+      calendarBeginId = ((DiscussionForum) forumItem).getCalendarBeginId();
+      calendarEndId = ((DiscussionForum) forumItem).getCalendarEndId();
+      sendOpenCloseToCalendar = ((DiscussionForum) forumItem).getSendOpenCloseToCalendar();
+      openDate = ((DiscussionForum) forumItem).getOpenDate();
+      closeDate = ((DiscussionForum) forumItem).getCloseDate();
+      membershipItems = ((DiscussionForum) forumItem).getMembershipItemSet();
+      availabilityRestrictedNow = ((DiscussionForum) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else {  //don't do anything with forumItem or the Calendar if it's not a Forum or a Forum Topic.
+      return;
+    }
+    Collection<Group> allowedGroups = getAllowedGroups(membershipItems);
+    if (allowedGroups.size() == 0 && !studentsAllowed(membershipItems)){
+      sendOpenCloseToCalendar = false; //if no groups or students are allowed to see this, we will treat it as though Send is not checked at all.
+      if (forumItem instanceof DiscussionTopic){  //clear the forum item's value as well, if this is the case.
+        ((DiscussionTopic) forumItem).setSendOpenCloseToCalendar(false);
+      } else if (forumItem instanceof DiscussionForum) {
+        ((DiscussionForum) forumItem).setSendOpenCloseToCalendar(false);
+      }
+    }
+    try {   //now actually start processing the data for Calendar.
+      Calendar targetCalendar = this.calendarService.getCalendar("/calendar/calendar/" + getContextSiteId().replace("/site/", "") + "/main");
+      CalendarEventEdit begin = null;
+      CalendarEventEdit end = null;
+      if(sendOpenCloseToCalendar && availabilityRestrictedNow){ //when Send To Calendar is checked and the availability is date-restricted in some way
+        if (calendarBeginId != null){ //if there is already a Calendar record for the forum opening
+          try {
+            begin = targetCalendar.getEditEvent(calendarBeginId, CalendarService.EVENT_MODIFY_CALENDAR);
```

Comment:
Are you sure targetCalendar is non null here? Can it ever be null? AssignmentServiceImpl checks for IdUnusedException while getting the calendar, so perhaps it can be.

Reply:
It looks like it is possible for it to be null, thanks for catching that! I'll follow it with this:
`if(targetCalendar == null){
        return;
      }`